File: sorting/bubble.py
# time complexity of the bubble sort algorithm is O(n^2)....worst case scenario ( i guess 😂)
# This is because the algorithm has to run through the array
def bubble_sort(arr):
    n = len(arr)
    for i in range(n):  # This is to run the process multiple times
        print(f"interation [{i}] : arr - > {arr}")
        # its n-i-1 because we dont need to compare the last element
        for j in range(0, n - i - 1):  # this is for comparing and doing the swapping
            # if the current element is greater than the next element
            if arr[j] > arr[j + 1]:
                # swapping the elements
                arr[j], arr[j + 1] = arr[j + 1], arr[j]

    return arr


val = [64, 34, 25, 12, 22, 11, 90]
print(bubble_sort(val))

File: sorting/bubble.cpp
#include <iostream>
#include <vector>

// bubble sort algorithm to sort the array
//
void bubble_sort(std::vector<int> &arr) {
  // just loop and exchange it basically
  for (int i = 0; i < arr.size(); i++) {
    for (int j = 0; j < arr.size(); j++) {
      if (arr[j] < arr[j + 1]) {
        arr[j], arr[j + 1] = arr[j + 1], arr[j];
        // int temp = arr[j];
        // arr[j] = arr[j + 1];
        // arr[j + 1] = temp;
      }
    }
  }
  // return array.. just for you to see
  for (int i = 0; i < arr.size(); i++) {
    std::cout << arr[i] << " ";
  }
}

int main() {
  std::vector<int> arr = {5, 6, 7, 8, 9, 0};
  bubble_sort(arr);
  return 0;
}
